lib/paper_trail/attribute_serializers/cast_attribute_serializer.rb

```diff
@@ -44,7 +44,13 @@ def deserialize(attr, val)
       end
 
       def serialize(attr, val)
-        AttributeSerializerFactory.for(@klass, attr).serialize(val)
+        serialized = AttributeSerializerFactory.for(@klass, attr).serialize(val)
+        case serialized
+        when Date, Time
+          @klass.connection.type_cast(serialized)
+        else
+          serialized
+        end
```

Comment:
This seems similar to `TypeSerializers::PostgresArraySerializer`, in that it wraps a particular type (in this case `Date, Time`). Should we follow that pattern, and have a `TypeSerializers::DateTimeSerializer`?